src/db.rs

```diff
@@ -1035,14 +1107,21 @@ impl DB {
         }
     }
 
-    pub fn merge_cf_opt(
+    pub fn merge_cf_opt<K, V>(
         &self,
         cf: ColumnFamily,
-        key: &[u8],
-        value: &[u8],
+        key: K,
+        value: V,
         writeopts: &WriteOptions,
-    ) -> Result<(), Error> {
+    ) -> Result<(), Error>
+        where K: AsRef<[u8]>, 
+              V: AsRef<[u8]> {
+        
+        let key = key.as_ref();
+        let value = value.as_ref();
```

Comment:
Is it necessary ? What if just write below:
```rust
key.as_ref().as_ptr() as *const c_char,
``` 
???

Reply:
The point is less lines of code is better 🙂

Reply:
Only because I'd have to do it twice on lines 1129 and 1130. Either way works I suppose...